opentelemetry-api/src/opentelemetry/trace/__init__.py

```diff
@@ -102,6 +103,34 @@ def get_context(self) -> 'SpanContext':
             A :class:`.SpanContext` with a copy of this span's immutable state.
         """
 
+    def set_attribute(self: 'Span',
+                      key: str,
+                      value: 'types.AttributeValue'
+                      ) -> None:
+        """Sets an Attribute.
+
+        Sets a single Attribute with the key and value passed as arguments.
+        """
+
+    def add_event(self: 'Span',
+                  name: str,
+                  attributes: 'types.Attributes',
+                  ) -> None:
+        """Adds an Event.
```

Comment:
One open question is whether the caller ever needs access to the events or links they create (in which case we need the classes in the API) or if we should treat them as part of the implementation of these methods.

Reply:
There are two places where I've seen `Event`/`Link` and related members being publicly used/exposed : 1) `SpanData` (but as that is probable going to the SDK at the very least) and 2) `MessageEvent` (which extends `Event`, at least in Java: https://github.com/open-telemetry/opentelemetry-java/blob/master/contrib/trace_utils/src/main/java/io/opentelemetry/contrib/trace/MessageEvent.java).

Makes me *feel* we should include these `Event`, etc classes here.